### PR TITLE
Feature/cit 455 finalize azure roundtrip

### PR DESCRIPTION
**Pull request by abelikt**
## Proposed changes

Finally add a roundtrip test for Azure IoT.
- Add script to upload certificate
- Add roundtrip test code
- Add Azure smoke test script
- Update workflows 
- Extract common code to configure bridge

TODO : We need to decide if we completely drop the old service bus trick.

Replacement for https://github.com/thin-edge/thin-edge.io/pull/287

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://cumulocity.atlassian.net/browse/CIT-455

## Checklist

- x ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by makr11st — APPROVED** [review]
I think it looks ok, however, it would be good if we could have a little more documentation about the steps performed.

File: ci/az_upload_device_cert.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+
+def delete_device(devname, hub, sas_name):
+    """Delete the device"""
+
+    if "SASKEYIOTHUB" in os.environ:
+        sas_policy_primary_key_iothub = os.environ["SASKEYIOTHUB"]
+    else:
+        print("Error environment variable SASKEYIOTHUB not set")
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (abelikt) [reply ci/az_upload_device_cert.py]: ...upated all the places to be more pythonic

**Review by PradeepKiruvale — APPROVED** [review]
LGTM, 
I feel we can remove queue based testing, can just retain the SDK based one.

File: ci/az_upload_device_cert.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        print("Device Properties: ", req.text)
+    if req.status_code == 204:
+        print("Unconditionally deleted the device")
+        print("Deleted Device Properties: ", req.text)
+    elif req.status_code == 404:
+        print("Device is not there, not deleted")
+    else:
+        print(f"Error: {req.status_code}")
+        print("Response Properties", req.text)
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (abelikt) [reply ci/az_upload_device_cert.py]: I added a raise_for_status, but I still need to filter that 404 out. That will happen when the certificate is new

File: .github/workflows/integration-test-workflow.yml [hunk 3]
```diff
@@ -181,7 +184,7 @@ jobs:
       - name: chmod dummy_plugin
         run: chmod +x /home/pi/tedge_dummy_plugin/tedge_dummy_plugin
 
-      - name: Run Smoke Test
+      - name: Run smoke test for Cumulocity
         run: ./ci/ci_smoke_test.sh
```
[the remark this answers is not in the document]
Comment (abelikt) [reply .github/workflows/integration-test-workflow.yml]: Done

**Pull request merged**